tools/units.py

```diff
@@ -145,18 +145,26 @@ class Units:
     dicts['stress'] = stress
 
 
+    # Precompute a dict mapping lowercased unit names to quantityName:conversionFactor pairs
+    quantities_for_unit = {}
+    for quantity in dicts:
+        for unit, factor in dicts[quantity].items():
+            unit = unit.lower()
+            if unit not in quantities_for_unit:
+                quantities_for_unit[unit] = {}
+            quantities_for_unit[unit][quantity] = factor
+
+
     def __init__(self):
         raise UnitsError('Instances of Units cannot be created')
 
 
     @classmethod
     def find_unit(cls, unit):
-        ret = {}
-        for quantity in cls.dicts:
-            for k in cls.dicts[quantity]:
-                if k.lower() == unit.lower():
-                    ret[quantity] = k
-        return ret
+        try:
+            return cls.quantities_for_unit[unit.lower()]
+        except KeyError:
+            return {}
```

Comment:
```suggestion
        return cls.quantities_for_unit.get(unit.lower(), {})
```
In principle you could make this a bit more compact by using `dict.get()` instead of a `try`/`except` approach.

Reply:
Of course, that's way better. I think it's obvious by now that I'm not that much of a Python expert.

I have also just realized that `find_unit()` doesn't have a leading underscore, so I likely shouldn't have changed its return type. I'll fix that shortly by calling `get()` directly in `conversion_ratio()` and making a separate `find_unit()` that maintains the original signature.